client_test.go

```diff
@@ -352,6 +353,128 @@ func TestClient_Do_WithResponseHandler(t *testing.T) {
 	}
 }
 
+func TestClient_Do_WithPrepareRetry(t *testing.T) {
+	// Create the client. Use short retry windows so we fail faster.
+	client := NewClient()
+	client.RetryWaitMin = 10 * time.Millisecond
+	client.RetryWaitMax = 10 * time.Millisecond
+	client.RetryMax = 2
+
+	var checks int
+	client.CheckRetry = func(_ context.Context, resp *http.Response, err error) (bool, error) {
+		checks++
+		if err != nil && strings.Contains(err.Error(), "nonretryable") {
```

Comment:
Alternatively we can use here `!errors.Is(err, retriableErr)`. or sth like that. Don't know if it's not an overkill tho

Reply:
I prefer to keep it in sync with the original test